### PR TITLE
[REFACTOR] Remove unnecessary conditionals from code

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?

Fewer conditionals makes the code easier to reason about.

### CODE REVIEW

File: spec/services/cocina/mapping/descriptive/h2_datacite/title_h2_datacite_spec.rb [hunk 1]
```diff
@@ -20,56 +20,11 @@
...
-      expect(title_attributes).to eq(
+      expect(title_attributes).to eq [
         {
           title: 'Tales of a brooding sea star'
         }
-      )
-    end
-  end
-
-  ### --------------- specs below added by developers ---------------
```
Comment (jcoyne) [inline spec/services/cocina/mapping/descriptive/h2_datacite/title_h2_datacite_spec.rb:36]: These are not possible https://github.com/sul-dlss/cocina-models/blob/main/openapi.yml#L536

File: app/services/cocina/to_datacite/dro_access.rb [hunk 2]
```diff
@@ -15,13 +15,13 @@ def initialize(cocina_item_access)
         @cocina_item_access = cocina_item_access
       end
 
-      # @return [Hash] Hash of DataCite rightsList attributes, conforming to the expectations of HTTP PUT request to DataCite
+      # @return [Array<Hash>] list of DataCite rightsList attributes, conforming to the expectations of HTTP PUT request to DataCite
```
Comment (justinlittman) [inline app/services/cocina/to_datacite/dro_access.rb:18]: or NilClass

**Review by justinlittman — APPROVED** [review]
On minor suggested change.

File: app/services/cocina/to_datacite/dro_access.rb (hunk 2 above)
Comment (jcoyne) [reply app/services/cocina/to_datacite/dro_access.rb]: Yes. Good catch. I have fixed it.

**Pull request merged**